tidy3d/components/source.py

```diff
@@ -291,7 +294,83 @@ def amp_time(self, time: float) -> complex:
         return const * offset * oscillation * amp
 
 
-SourceTimeType = Union[GaussianPulse, ContinuousWave]
+class CustomSourceTime(Pulse):
+    """Custom source time dependence.
```

Comment:
one thing also that might be worth mentioning explicitly is whether this can be complex-valued.